eml_parser/eml_parser.py

```diff
@@ -669,37 +670,61 @@ def string_sliding_window_loop(body: str, slice_step: int = 500, max_distance: i
                 ptr_start = ptr_end
 
     @staticmethod
-    def get_uri_ondata(body: str) -> typing.List[str]:
+    def get_uri_ondata(body: str, include_href: bool = True, email_force_tld: bool = False) -> typing.List[str]:
         """Function for extracting URLs from the input string.
 
         Args:
             body (str): Text input which should be searched for URLs.
+            include_href (bool): Include potential URLs in HREFs matching non-simple regular expressions
+            email_force_tld (bool): Exclude common file extensions that are not valid TLDs
 
         Returns:
             list: Returns a list of URLs found in the input string.
         """
         list_observed_urls: typing.Counter[str] = Counter()
 
-        for found_url in eml_parser.regexes.url_regex_simple.findall(body):
-            if '.' not in found_url:
+        def clean_found_uri(url: str) -> typing.Optional[str]:
+            if '.' not in url and '[' not in url:
                 # if we found a URL like e.g. http://afafasasfasfas; that makes no
-                # sense, thus skip it
-                continue
+                # sense, thus skip it. Include http://[2001:db8::1]
+                return
 
             try:
-                found_url = urllib.parse.urlparse(found_url).geturl()
+                url = url.lstrip('"\'\t \r\n').replace('\r', '').replace('\n', '')
+                url = urllib.parse.urlparse(url).geturl()
+                if email_force_tld:
+                    scheme_url = url
+                    if ':/' not in scheme_url:
```

Comment:
shouldn't this be **://** ?

Reply:
It should, otherwise the netloc from urlparse is incorrect.
The URL regex matches on `:/`, `://`, and `:///`. I need to clean the matches more before parsing with urllib.